### ISSUE TITLE
Cannot update record if new format of accessor is used (Column not found: 1054 Unknown column)

### ISSUE THREAD

**Issue by RoodFruit** [laravel/framework#40470]
- Laravel Version: 8.78.1
- PHP Version: 8.1.1
- Database Driver & Version: MySQL 5.7.34

### Description:
I'm using the new format for defining accessors and mutators. After that change, every time I run a command that updates the status of something in the database **based on an accessor** in suddenly get the error:
```
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'ends_at' in 'field list' (SQL: update `example_table` set `status` = active, `ends_at` = ?, `example_table`.`updated_at` = 2022-01-18 10:05:37 where `id` = 481)
```

As you can see, the `ends_at` field has been added, but this doesn't exist in the table. The reason this gets added, is that I call the accessor to do some check right before the `save()`. What I found out it is that the accessorCast array is filled in the model. Which is the cause why its added to the query.

Here is the part of the Command (reducted):
```
<?php

namespace App\Console\Commands;

use ...

class ExampleCommand extends Command
{
    protected $signature = '...';

    protected $description = '...';

    public function __construct()
    {
        parent::__construct();
    }

    public function handle(): int
    {
        $exampleTableModels = ExampleTableModel::query()
                                     ->whereIn('status', ['...'])
                                     ->get();

        $now = CarbonImmutable::now();

        foreach ($exampleTableModels as $exampleTableModel) {
            /** @var CarbonInterface $startsAt */
            $startsAt = $exampleTableModel->starts_at;
            
            /** @var CarbonInterface $endsAt */
            $endsAt = $exampleTableModel->ends_at;    <--- THIS CAUSES THE ERROR!

            // If end date has been reached
            if ($endsAt->lte($now)) {
                $exampleTableModel->status = '...';
                $exampleTableModel->save();

                continue;
            }
        }

        return Command::SUCCESS;
    }
}

```

And this is the model definition (reducted):
```
<?php

namespace App\Models;

use ...

class ExampleModel
{
    protected $table = 'example_table';

    protected $fillable = [
        'starts_at',
        'period_in_months',
        'status',
        'created_at',
        'updated_at',
    ];

    protected $appends = [
        'ends_at'
    ];

   
    protected $casts = [
        'starts_at'        => 'datetime',
        'period_in_months' => 'integer',
        'status'           => 'string',
        'created_at'       => 'datetime',
        'updated_at'       => 'datetime',

        // Derivatives (accessors)
        'ends_at' => 'datetime'
    ];


    /*
    |--------------------------------------------------------------------------
    | Accessors / Mutators
    |--------------------------------------------------------------------------
    */

    /**
     * Get ends_at.
     */
    protected function endsAt(): Attribute
    {
        return new Attribute(
            get: function (): CarbonInterface {
                /** @var CarbonInterface $startsAt */
                $startsAt       = $this->starts_at;
                $periodInMonths = $this->period_in_months;

                return $startsAt->addMonths($periodInMonths);
            }
        );
    }
}

```

### Steps To Reproduce:

1. Make make with accessor.
2. Run a command where you access the accessor (just a get is fine).
3. Update the model.

**Comment by driesvints** [comment]
This was fixed in Laravel v8.79.0: https://github.com/laravel/framework/commit/29a6692fb0f0d14e5109ae5f02ed70065f10e966

**Comment by RoodFruit** [comment]
@driesvints awesome, thanks!
